### ISSUE TITLE
Windows version

### ISSUE THREAD

**Issue by sebastien-bartoli** [spolu/breach_core#90]
Awesome project, I hope it'll get traction and be popular ! Anyhow, a Windows version would be really nice. Is it on the tracks ? Is there some particular issues to work around to get it to work on Windows ?

**Comment by tudo75** [comment]
+1

**Comment by kurtextrem** [comment]
Would be nice, as many devs are on windows! If there is anything we can do to get a windows build up, tell us.

**Comment by luaks** [comment]
In breach/exo_browser#142 @spolu explains, that when aura is used instead of gtk, this will probably be a lot easier to implement.

**Comment by spolu** [comment]
@derLukers exactly.

**Comment by JasonKraft** [comment]
I look forward to that eventual release :+1: :)

**Comment by maxigs7** [comment]
+1

**Comment by zackp30** [comment]
:+1: Would be useful to see a Windows build. :smile:

**Comment by Myrannas** [comment]
Would love to see a windows build, would really unlock some interesting development opportunities for this platform (eg easy multi platform desktop apps).

**Comment by vinz243** [comment]
:+1:

**Comment by aozora** [comment]
+1

**Comment by ghost** [comment]
Well, node works on Windows as does Chromium so I imagine it can't be _that_ hard to make it work on Windows. It just needs someone to actually do it.

**Comment by spolu** [comment]
There is a thin layer of code for the window management that is platform specific. We just need to get that part done.

**Comment by nicolas-t** [comment]
:+1:

**Comment by Atrox** [comment]
:+1:

**Comment by amq** [comment]
+1

**Comment by sriharshav** [comment]
:+1:

**Comment by davisable** [comment]
:+1:

**Comment by diegowifi** [comment]
I agree, it would be really awesome to have this available for Windows :smile:

**Comment by thecotne** [comment]
+1

**Comment by devon** [comment]
+1

**Comment by macleod-ee** [comment]
+1

**Comment by CREEATION** [comment]
:+1:

**Comment by action-simon** [comment]
+1

**Comment by ivernaloo** [comment]
+1

**Comment by alexhulbert** [comment]
:+1: This is the coolest browser I've ever seen 0_o. It's so sleek-looking.

**Comment by thecotne** [comment]
+1 for windows version

**Comment by batjko** [comment]
I think there've been enough +1s now. It would be cool though if there was a roadmap or timeline to at least some early self-build windows version or something along those lines.

**Comment by cyrusdavid** [comment]
:+1:

**Comment by spolu** [comment]
I'll release very soon a road map of the major changes to come for Breach and ExoBrowser. 
Windows support will come out of the box in the next version of the ExoBrowser... 

I'll file some issue as well to keep track of progress. We definitely want windows support ASAP!

**Comment by ThamiduM** [comment]
![](https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png)

**Comment by mufid** [comment]
Is there any ETA for Windows build?

**Comment by spolu** [comment]
Windows support should be available with exo_browser v0.7 (based on Aura). Track https://github.com/breach/exo_browser/issues/161 for updates!

**Comment by sandeshdamkondwar** [comment]
:+1:

**Comment by Antonin-Deniau** [comment]
:+1:

**Comment by ghost** [comment]
Windows version would be awesome, please make it possible!

**Comment by davisonio** [comment]
:+1:

**Comment by spolu** [comment]
Working on it! We're making good progress here: https://github.com/breach/thrust/issues/161

Once Thrust (ex ExoBrowser) v0.7 will be released we'll focus on a bringing Breach to Windows!

**Comment by solocshaw** [comment]
Great idea.

**Comment by spolu** [comment]
Initial windows support of http://github.com/breach/thrust
![screenshot from 2014-10-21 07 42 44](https://cloud.githubusercontent.com/assets/15067/4720397/ccd03c48-5930-11e4-8647-5e49bf445691.png)

This stuff take time guys, sorry for the wait. We're working hard on this!

**Comment by sebastien-bartoli** [comment]
That is wonderful. It's great to see you keeping us updated !

**Comment by spolu** [comment]
:+1:  :)

**Comment by davisable** [comment]
@spolu Great to see progress! Looking forward to it

**Comment by spolu** [comment]
Thanks! You can start having fun with https://github.com/breach/thrust in the meantime in case any of you guys in intersted in using an cross-platform application-shell bindable from any language (nodeJS, Go, Python, ...)

**Comment by ghost** [comment]
Hehe.... Cool to see you guys making more and more progress! Go windows version :+1:

**Comment by whatsdis** [comment]
do we have a date on the windows release?

Really realy really would like this on windows! I can't wait!!!

**Comment by spolu** [comment]
Targeting before Christmas but Breach might be substantially different in its next release...
You'll have access to development versions before then!

**Comment by 5nyper** [comment]
Link leads to Gandi.net, is there a new link?

**Comment by cxdy** [comment]
Can't wait for this. Loved it on my Linux box and can't wait to see it running on my laptop. :)

**Comment by spolu** [comment]
@cxdy working on it! :) Feel free to test out http://github.com/breach/thrust in the meantime. Would love to get your comment on it.

**Comment by spolu** [comment]
@Ap0ph1s Which link?

**Comment by cxdy** [comment]
Oh, sweet. I'll check it out when I get home. Thanks :)

**Comment by 5nyper** [comment]
http://thrust.ws

**Comment by cxdy** [comment]
Oh, yeah, that does link to something else.

**Comment by spolu** [comment]
Yep there's no webpage set up there yet. It'll be soon.

**Comment by rclai** [comment]
WINDOWZ ZUPPORT PLZ! :thumbsup: :thumbsup:  :thumbsup:  :thumbsup:  :thumbsup:  :thumbsup:  :thumbsup: :scream: :scream: :scream: :scream: :scream: :scream: :scream:

**Comment by ghost** [comment]
Agreed, i really want to change finally browsers, so we need Windows
support 8-D

2014-11-09 23:48 GMT+01:00 rclai notifications@github.com:

> WINDOWZ ZUPPORT PLZ! [image: :thumbsup:] [image: :thumbsup:] [image:
> :thumbsup:] [image: :thumbsup:] [image: :thumbsup:] [image: :thumbsup:] [image:
> :thumbsup:] [image: :scream:] [image: :scream:] [image: :scream:] [image:
> :scream:] [image: :scream:] [image: :scream:] [image: :scream:]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/breach/breach_core/issues/90#issuecomment-62323725.

## 

---------- Wikus Van De Merwe was right - Fookin' Planet!!! ----------

**Comment by cxdy** [comment]
Hey @spolu any guides on how to use Thrust with Breach?

**Comment by spolu** [comment]
@cxdy Thrust is what will power Breach going forward. You can use it to create cross-platform native apps using NodeJS or Go (and other language coming soon). You should have enough to get started with it here: http://github.com/breach/thrust You can also check the tests here: https://github.com/breach/node-thrust/tree/master/test

**Comment by cxdy** [comment]
@spolu Oh cool, I'll check it out.

**Comment by spolu** [comment]
If you want to have fun check out JankyBrowser by @morganrallen. The cross-platform Browser that fits in a Gist! https://gist.github.com/morganrallen/f07f59802884bcdcad4a

**Comment by jahlomp** [comment]
+1

**Comment by ericrange** [comment]
+1

**Comment by 5nyper** [comment]
Any updates?

**Comment by cxdy** [comment]
+1

**Comment by ericrange** [comment]
+1

**Comment by spolu** [comment]
Still working on it. We're tweaking the thrust platform to fit the needs of Breach (finalize webview tag, upgrade to chrome 39, request proxying) and we'll then work on an initial version for Breach.

Anybody who would feel like helping on this, please let me know :)

Out of curiosity, what is motivating you to try out Breach?

**Comment by cxdy** [comment]
What skillset is needed to contribute? I'd love to help.

Also, the expandibility of Breach seems awesome. It also looks sick from the site, and I've used it on Linux and loved it.

**Comment by 5nyper** [comment]
I want to use it because I can install nose modules, and the fact that I
used Node event loops makes me want to try it ou, especially how
customizable it is.

**Comment by spolu** [comment]
@cxdy C++ programming would be super-helpful for Thrust. Building version on windows is also really helpful. Come chat on Freenode on #breach 

@cxdy @Ap0ph1s would you like to customize it by picking modules or by coding modules of your own?

**Comment by cxdy** [comment]
@spolu Afraid I don't know too much C++ yet! I have to learn it.

I would pick modules and code them myself to customize Breach.

**Comment by sebastien-bartoli** [comment]
@spolu I think the greatness of Breach comes from the fact that it's built on top of chromium (and therefore the blink rendering engine), and its hackability. As a webdeveloper, it would be the perfect tool for me to hack my browser the same way I hack my code editor :)

I'm not skilled in C++ but if you need any JS help on any part of the project, I'll gladly help :)

**Comment by spolu** [comment]
OK! Will make sure to ping you guys for any help on the JS front

**Comment by 5nyper** [comment]
Both code my own and install modules, its just fun to play with clay :)

**Comment by ghost** [comment]
What is motivating me wanting to use Breach.... Firefox has become a crap
browser, a follower instead of a leader and a traitor of the open source
idea... so Breach sounds like a worthy replacement :D

2014-12-01 17:48 GMT+01:00 Stanislas Polu notifications@github.com:

> Still working on it. We're tweaking the thrust platform to fit the needs
> of Breach (finalize webview tag, upgrade to chrome 39, request proxying)
> and we'll then work on an initial version for Breach.
> 
> Anybody who would feel like helping on this, please let me know :)
> 
> Out of curiosity, what is motivating you to try out Breach?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/breach/breach_core/issues/90#issuecomment-65094458.

## 

---------- Wikus Van De Merwe was right - Fookin' Planet!!! ----------

**Comment by alexhulbert** [comment]
@SJ-Diamond Couldn't have said it better myself. I still respect FireFox, but I'm really drawn to the idea of being able to pimp out my browser _my_ way rather than just going with whatever Chrome/FF has to offer.

**Comment by cxdy** [comment]
@Taconut @SJ-Diamond Exactly.
I hate being limited to whatever they have in the Chrome and Firefox stores. 
You can install extensions in developer mode on both browsers but they're hard to find.
I like that you can design the browser how you want it.

**Comment by ghost** [comment]
Well, about the minimalistic feature set of Chrome and it being Anti-Power
user... there are no additional words necessary, but it is incredible
shocking that Mozilla decided that their future user group are the same
users like Google Chrome, so naturally they cut all away which is
Pro-Poweruser.

And at least i find that.... well, disgusting!

2014-12-03 21:27 GMT+01:00 Alex Hulbert notifications@github.com:

> @SJ-Diamond https://github.com/SJ-Diamond Couldn't have said it better
> myself. I still respect FireFox, but I'm really drawn to the idea of being
> able to pimp out my browser _my_ way rather than just going with whatever
> Chrome/FF has to offer.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/breach/breach_core/issues/90#issuecomment-65484161.

## 

---------- Wikus Van De Merwe was right - Fookin' Planet!!! ----------

**Comment by sebastien-bartoli** [comment]
On a completely unrelated note, I could definitely see a breach-modules github organisation where people could push their repos of breach mods they did. 

Like, you want to share an awesome breach extension you coded ? Push it on github !

**Comment by cxdy** [comment]
@sebastien-bartoli They already have that. Or are you talking about making a repository that you could push your module into?

Anyway, [check it out](https://github.com/breach/breach_core/wiki/List-of-modules).

**Comment by sebastien-bartoli** [comment]
Oh yes. This exists :) I'm late to the battle

**Comment by cxdy** [comment]
😉

From: Sebastien Bartoli
Sent: ‎Thursday‎, ‎December‎ ‎4‎, ‎2014 ‎10‎:‎11‎ ‎AM
To: breach/breach_core
Cc: Cody Kaczynski

Oh yes. This exists :)

**Comment by 5nyper** [comment]
Anyyy Update?

**Comment by ericrange** [comment]
+1

**Comment by matbeedotcom** [comment]
+100 :)

**Comment by ericrange** [comment]
+1000

**Comment by vinz243** [comment]
+10000 :)

**Comment by oskardahlberg** [comment]
+10

**Comment by ericrange** [comment]
+1000000000^100000

**Comment by sebastien-bartoli** [comment]
Guys, seriously, this does not help and just spam everybody inboxes. Please only comment if you have something more interesting to say than "+1" or whatever.

**Comment by cxdy** [comment]
+1

Agreed, the amount of emails I'm getting over this is ridiculous.

**Comment by morganrallen** [comment]
we are accepting contribution.

**Comment by spolu** [comment]
Hi guys. A little update on that. I haven't had all the time I wanted in the past weeks. 

Thrust which was the new techno stack for breach was intended to be a chromium shell that supports webviews. Doing so we incited the folks at atom-shell to add webviews as well. So it'll be better to join forces with them. Bottom line, thrust will probably become an atom-shell wrapper so that it can be used with vanilla node / io.js. This does not require a lot of work as we already had such prototype months ago. 

One last thing that is needed from atom-shell to implement a proper browser is the ability to manage cookies and sessions. This requires quite more work and is tracked here: https://github.com/atom/atom-shell/issues/959. Once we land that (and I don't know when it'll be) we can go back to breach and refactor it on top of this new stack.

I'm pretty busy right now unfortunately, so I can't make any promises about when it'll be. Breach is therefore going in cryogenic statis. But it'll eventually wake up (esp. if you contribute!). We all want different stuff from a browser, we've seen it with Vivaldi. It's exciting to see new implementations, but it also feel like we'd like to build our own stuff. It seems that one size does not fit all and that's why Breach is exciting and will eventually come back!

Hope it helps.
Up and to the moon.

**Comment by vinz243** [comment]
+1 It's spamming my webmail

**Comment by ericrange** [comment]
+1 Yes it does!

**Comment by dittoslash** [comment]
+15, going to check Thrust

**Comment by ericrange** [comment]
+100 @dittoslash dont spam!! +100

**Comment by vinz243** [comment]
My inbox is full of Windows Version now!!

**Comment by ericrange** [comment]
+1000 wtf?? :open_mouth:

**Comment by 5nyper** [comment]
Alright, we dont need anymore '+1'|'+100", The developers already know that we all need one.

**Comment by action-simon** [comment]
One more +532895892375892376982376 comment and I never will come back to test/support this project. You do harm this project with spamming guys. And yes sorry I know my comment is not needed as well but when I get hundreds of mails with only senseless spam it's going on my nerves...

**Comment by ericrange** [comment]
so just build the damn windows binaries and do your other stuff later on? after this you will NEVER get this "+1" emails again?!?!

**Comment by tudo75** [comment]
if only 10% of +1 people was involved in the development i think all was build
people who work on this porject do it in **their** spare time and this is **their** project so **they** plan the workflow and what to do and when
if someone is hurry he can only start to contribute actively to the project and don't cry as a child to have a new toy, alternatively stop to waste thei time and wait
**Thank to the contributors for all their work**

**Comment by vohof** [comment]
> +1000 wtf??

> so just build the damn windows binaries and do your other stuff later on? after this you will NEVER get this "+1" emails again?!?!

Oh wow.

**Comment by mastilver** [comment]
You can also contribute with a bounty:
https://www.bountysource.com/issues/3077134-windows-version

**Comment by gliurn** [comment]
Can you explain what you re doing with thrust right know on http://breach.cc/? because i spend some days to search after the evolution of the project without knowing it was stop

**Comment by jeromeyers** [comment]
I'd like to contribute, C# or JS.

**Comment by cxdy** [comment]
Looks like he needs help with the C++ end @jeromeyers  https://github.com/breach/breach_core/issues/90#issuecomment-65096106

**Comment by SgtPooki** [comment]
@bradgearon, you should come check this out...

**Comment by spaghetticodr** [comment]
+9001

**Comment by cxdy** [comment]
Any news so far?
